### PR TITLE
findpeaks: fix check for dists < MinPeakDistance

### LINKED ISSUE #18: control flow logic in findpeaks
https://github.com/gjmvanboxtel/gsignal/blob/0fbb9b185a353df081a46e604f14a8eee6c5ca36/R/findpeaks.R#L173

This should be 

```
if (any(D<MinPeakDistance)) {
```

because otherwise it is just checking if D has some positive values rather than comparing distances with `MinPeakDistance`.

Compare with the corresponding line in octave

https://sourceforge.net/p/octave/signal/ci/default/tree/inst/findpeaks.m#l205

```
if (any (D(:) < minD))
```

### PR DESCRIPTION
**Pull request by jefferis**
* fixes https://github.com/gjmvanboxtel/gsignal/issues/18

### CODE REVIEW

**Comment by jefferis** [comment]
cc @sfrechter

File: R/findpeaks.R [hunk 1]
```diff
@@ -170,7 +170,7 @@ findpeaks <- function(data,
   D <- with(expand.grid(A = idx_s, B = t(idx_s)), abs(A - B))
   dim(D) <- c(length(idx_s), length(idx_s))
   diag(D) <- NA                     # eliminate diagonal comparison
-  if (any(D) < MinPeakDistance) {
+  if (isTRUE(any(D < MinPeakDistance))) {
```
Comment (jefferis) [inline R/findpeaks.R:173]: compare with https://sourceforge.net/p/octave/signal/ci/default/tree/inst/findpeaks.m#l205

Comment (jefferis) [reply R/findpeaks.R:173]: NB isTRUE handles cases when there are only NA values in D

**Comment by gjmvanboxtel** [comment]
Done, I am currently abroad with limited internet access. Hope it works out for you

**Comment by jefferis** [comment]
Oh thanks so much — especially in the circumstances — and more generally for your contribution to the community.

**Pull request merged**